### ISSUE TITLE
Dependency is not registreted when namespace is non-standart named

### ISSUE THREAD

**Issue by deepj** [dry-rb/dry-rails#20]
**Describe the bug**

I have a namespace which looks like (example) `CloudNATS`. This is a namespace of Rails application declared in `config/application.rb` as well.

When I tried to import a dependecy, I got

```
Dry::Container::Error (Nothing registered with the key :http)
```

**To Reproduce**

```ruby
# app/operations/create_user
class CreateUser
  include CloudNATS::Import[:http]
end
```

```ruby
# lib/cloudnats/http.rb
module CloudNATS
  class HTTP
  end
end
```

**Expected behavior**

The dependency is successfully imported without any exception. My namespace would 

**Your environment**

- Affects my production application: **NO**
- Ruby version: 2.7.1
- OS: macOS

**Comment by solnic** [comment]
This is expected since it's not a regular word that includes an acronym that you need to register in your inflector. I'm not sure if registering just the acronym or the entire word will work, you need to verify that yourself.

**Comment by deepj** [comment]
The acronym is declared in `config/initializers/inflections.rb` 🤧

**Comment by solnic** [comment]
@deepj yeah but the path to files does not match the constant name, `lib/cloudnats/http.rb` should be `lib/cloud_nats/http.rb`. Whatever you do, you need to make sure that the inflector correctly converts the dir names into constant names.

**Comment by deepj** [comment]
@solnic The problem is, at least in my understanding the situation, the Rails' Inflector returns `cloudnats` for `CloudNATS`

```
Loading development environment (Rails 6.0.2.2)
irb(main):001:0> 'CloudNATS'.underscore
=> "cloudnats"
```

I tried `lib/cloud_nats/http.rb` as well, but without any success :(

**Comment by deepj** [comment]
I know why it's happening 🙀

![Screenshot 2020-04-21 at 10 52 09](https://user-images.githubusercontent.com/33736/79846640-3847d000-83ae-11ea-8205-21978d891dbb.png)

I guess I would update documentation because it's taken from `dry-rails` documentation 😹 

Suggestion, `dry-rails` may would expose `Zeitwerk` [inflection API](https://github.com/fxn/zeitwerk#inflection). Because after solving the above fix, I'm getting

```
expected file ~/lib/cloudnats/http.rb to define constant CloudNATS::Http, but didn't (Zeitwerk::NameError). 
```

And this is dangerous to set in Rails Inflector because it can affect 3rd-party gems which can use HTTP acronym in their names.

**Comment by solnic** [comment]
> Suggestion, dry-rails may would expose Zeitwerk inflection API

yeah I've thought about it. In general I would like to make dry-rails zeitwerk-only eventually, because it's a much better solution than whatever old Rails is doing.

**Comment by solnic** [comment]
> And this is dangerous to set in Rails Inflector because it can affect 3rd-party gems which can use HTTP acronym in their names. It's safer to use Zeitwerk inflection rather than Rails one.

I should add that you can (and in this case should) set up a custom inflector and configure it to your needs. It's a matter of doing `config.inflector = your_inflector`
